### PR TITLE
Fix errors in  Mapped pool and javadoc

### PR DESCRIPTION
**Pull request by gregw**
I just noticed this was wrong with our buffer pool javadoc

Signed-off-by: Greg Wilkins <gregw@webtide.com>

### CODE REVIEW

**Comment by gregw** [comment]
Also some real fixes now!

File: jetty-io/src/main/java/org/eclipse/jetty/io/MappedByteBufferPool.java [hunk 1]
```diff
@@ -302,4 +308,30 @@ public String toString()
             getMaxBucketSize(),
             getCapacityFactor());
     }
+
+    protected class Retained extends ArrayRetainableByteBufferPool
```
Comment (lachlan-roberts) [inline jetty-io/src/main/java/org/eclipse/jetty/io/MappedByteBufferPool.java:312]: Seems strange to me that you would create a `MappedByteBufferPool` and ask it for its `RetainableBufferPool` and it gives you an array based implementation.

Comment (gregw) [reply jetty-io/src/main/java/org/eclipse/jetty/io/MappedByteBufferPool.java:312]: yeah, but not sure we want to create a MappedRetainableByteBuffer pool.    Prior to the combination of pool, this is exactly what we would have got if the user selected a MBBP, as we would have created an ARBBP.   So just a bit strange that they are together.

I'm not actually sure if we really want to keep MBBP. @sbordet @lorban why do we have it?

Comment (lorban) [reply jetty-io/src/main/java/org/eclipse/jetty/io/MappedByteBufferPool.java:312]: AFAICT it's only used in the client, but I can't see why `MBBP` was created in the first place and I lack history. So unless @sbordet has a good reason to keep it, that pool impl can go IMHO.

Comment (gregw) [reply jetty-io/src/main/java/org/eclipse/jetty/io/MappedByteBufferPool.java:312]: OK, but for this PR, I'd like to fix javadoc and MBBP to unblock other work.
We can consider deprecating MBBP later.

Comment (lorban) [reply jetty-io/src/main/java/org/eclipse/jetty/io/MappedByteBufferPool.java:312]: Sounds good to me, and the change looks reasonable enough: currently MBBP is used together with a ARBBP so this change just solidifies a behavior that already exist.

Comment (gregw) [reply jetty-io/src/main/java/org/eclipse/jetty/io/MappedByteBufferPool.java:312]: @lorban can I get a green tick then please?

**Comment by gregw** [comment]
@lachlan-roberts @lorban Can I get this approved?

**Review by lorban — APPROVED** [review]

**Pull request merged**